### PR TITLE
_wip/3.3

### PR DESCRIPTION
**Pull request by z64**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,7 +5,84 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
...
+
+## [3.3.0] - TBD
+
+### Added
+
+- API methods to add and remove single member roles ([#310](https://github.com/meew0/discordrb/pull/310))
+- API methods and abstractions for listing available voice regions ([#311](https://github.com/meew0/discordrb/pull/311))
+- `Server` methods to prune members and to get the number of members available for pruning ([#315](https://github.com/meew0/discordrb/pull/315))
+- Methods for filtering the kinds of overwrites present on a channel ([#321](https://github.com/meew0/discordrb/pull/321))
+- `Channel#default_channl?`, for checking if a channel is the default channel of a server ([#320](https://github.com/meew0/discordrb/pull/320), thanks @Reaver01)
```
Comment (connorshea) [inline CHANGELOG.md]: Should be `Channel#default_channel?`

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,7 +5,84 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
...
+- Method for returning a `Server`'s `@everyone` role
+- Reaction can now be serialized with `#to_s` to be used in `Message#react` more easily ([#342](https://github.com/meew0/discordrb/pull/342))
+- Additional objects and attributes for parsing embeds on messages([#344](https://github.com/meew0/discordrb/pull/344), thanks @mattantonelli)
```
Comment (connorshea) [inline CHANGELOG.md]: Needs a space before the PR link.

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,7 +5,84 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
...
+
+### Changed
+
+- `Channel#make_invite` now accepts an argument to always return a unique invite code ([#312](https://github.com/meew0/discordrb/pull/312))
+- More of the API accepts objects that respond to `#resolve_id` ([#313](https://github.com/meew0/discordrb/pull/313), [#328](https://github.com/meew0/discordrb/pull/328), thanks @LikeLakers2)
+- `Channel#history` and `API::Channel.messages` now accepts `around_id` ([#314](https://github.com/meew0/discordrb/pull/314))
+- `API::Server.prune_count` accepts `days` ([#315](https://github.com/meew0/discordrb/pull/315))
+- Methods for creating channels accept additional arguments ([#321](https://github.com/meew0/discordrb/pull/321))
+- `Channel` overwrite-related API now returns an `Overwrite` object ([#321](https://github.com/meew0/discordrb/pull/321))
+- Creating roles now accepts more parameters([#323](https://github.com/meew0/discordrb/pull/323), thanks @Reaver01)
```
Comment (connorshea) [inline CHANGELOG.md]: Needs a space before the PR link

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,7 +5,84 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
...
+- Ratelimits are now logged to a `:ratelimit` logging level and can be configured
+- `client_id` in `Bot#initilalize` is now optional, and will be cached automatically by the API when needed ([#337](https://github.com/meew0/discordrb/pull/337))
+- `Voice::Encoder#encode_file` now accepts options for ffmpeg ([#341](https://github.com/meew0/discordrb/pull/341), thanks @oyisre)
+- Objects that implement `IDObject` can now be compared using more operators ([#346](https://github.com/meew0/discordrb/pull/346), thanks @mattantonelli)
+- Filled in permissions bit for viewing a server's audit log ([#349](https://github.com/meew0/discordrb/pull/349), thanks @Daniel-Worrall)
+- https://cdn.discordapp.com is now used as the base URL for CDN resources like avatars and server icons ([#358](https://github.com/meew0/discordrb/pull/358))
+- Reaction events raised from the bots actions will respect `parse_self` ([#350](https://github.com/meew0/discordrb/pull/350), thanks @Daniel-Worrall)
```
Comment (connorshea) [inline CHANGELOG.md]: should be `bot's actions`

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,7 +5,84 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
...
+- `default_channel` implementation is updated to reflect Discord changes ([#382](https://github.com/meew0/discordrb/pull/382))
+- Documentation around the conditions where our API returns `nil` is clarified ([#395](https://github.com/meew0/discordrb/pull/395), thanks @LikeLakers2)
+- Whenever possible, we update cached data about a `Server` returned to us from making changes to it
+- `Cache#server` now returns `nil` if a server is not found instead of raising an exception ([#424](https://github.com/meew0/discordrb/pull/424), thanks @soukouki)
+- `Bucket#rate_limited?` now accepts `increment` value for weighted rate limits ([#427](https://github.com/meew0/discordrb/pull/427), thanks @Lavode)
+- `Server#bans` now returns `Array<ServerBan>`, which contains `reason` data in addition to the user banned ([#404](https://github.com/meew0/discordrb/pull/404))
+- `Channel#pune` now accepts a block that can be used to filter the messages to be pruned ([#421](https://github.com/meew0/discordrb/pull/421), thanks @snapcase)
+
+### Dreprecated
```
Comment (connorshea) [inline CHANGELOG.md]: Should be `Deprecated`

File: CHANGELOG.md [hunk 6]
```diff
@@ -5,7 +5,84 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## Unreleased
-[View diff for this release.](https://github.com/meew0/discordrb/compare/v3.2.1...HEAD)
+[View diff for this release.](https://github.com/meew0/discordrb/compare/v3.3.0...HEAD)
+
+## [3.3.0] - TBD
```
Comment (connorshea) [inline CHANGELOG.md]: Needs `[3.3.0]: https://github.com/meew0/discordrb/releases/tag/v3.3.0` below this line so the link actually works once the tag exists.

File: CHANGELOG.md [hunk 7]
```diff
@@ -5,7 +5,84 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
...
+- `Cache#find_user` can now find individual users if name and discriminator is given ([#384](https://github.com/meew0/discordrb/pull/384))
+- `ReactionEvent` provides server, and member if possible ([#351](https://github.com/meew0/discordrb/pull/351), thanks @Daniel-Worrall)
+- Installation instructions now include guides for installing with bundler ([#386](https://github.com/meew0/discordrb/pull/386), [#405](https://github.com/meew0/discordrb/pull/405), thanks @VxJasonxV and @PixeLInc)
+- `default_channel` implementation is updated to reflect Discord changes ([#382](https://github.com/meew0/discordrb/pull/382))
+- Documentation around the conditions where our API returns `nil` is clarified ([#395](https://github.com/meew0/discordrb/pull/395), thanks @LikeLakers2)
+- Whenever possible, we update cached data about a `Server` returned to us from making changes to it
+- `Cache#server` now returns `nil` if a server is not found instead of raising an exception ([#424](https://github.com/meew0/discordrb/pull/424), thanks @soukouki)
+- `Bucket#rate_limited?` now accepts `increment` value for weighted rate limits ([#427](https://github.com/meew0/discordrb/pull/427), thanks @Lavode)
```
Comment (connorshea) [inline CHANGELOG.md]: Should be "accepts an `increment` value"

File: CHANGELOG.md [hunk 8]
```diff
@@ -5,7 +5,84 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
...
+- API support for managing webhooks ([#356](https://github.com/meew0/discordrb/pull/356), thanks @Daniel-Worrall)
+- Support for reading and managing a channel's `nsfw` property ([#380](https://github.com/meew0/discordrb/pull/380))
+- The `:administrator` permissions value is aliased as `:administrate` ([#322](https://github.com/meew0/discordrb/pull/322))
+- Class methods on `Permissions` for easily building permissions bits values ([#322](https://github.com/meew0/discordrb/pull/322))
+- `Gateway#send_packet` and `Gateway#send_raw` methods to send custom data payloads to the gateway
+- Methods for reading `icon_url` and `proxy_icon_url` in `EmbedAuthor`
+- Methods for obtaining a server and channels invites ([#394](https://github.com/meew0/discordrb/pull/394))
+- Example of using awaits ([#370](https://github.com/meew0/discordrb/pull/370))
+- Methods on `Member` for kicking and banning ([#404](https://github.com/meew0/discordrb/pull/404))
+- API method and abstraction for adding members to guilds with OAuth2 tokens ([#413](https://github.com/meew0/discordrb/pull/413/files))
```
Comment (connorshea) [inline CHANGELOG.md]: is it intentional that this links to the PR files page?

File: CHANGELOG.md [hunk 9]
```diff
@@ -5,7 +5,84 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
...
+### Fixed
+
+- `Errors::MessageTooLong` is now raised correctly ([#325](https://github.com/meew0/discordrb/pull/325), thanks @Daniel-Worrall)
+- Certain `Reaction` related events properly inherit `Event` ([#329](https://github.com/meew0/discordrb/pull/329), thanks @valeth)
+- Permissions calculation now takes the server's `@everyone` role permissions into consideration (additional work by [#357](https://github.com/meew0/discordrb/pull/357), thanks @mattantonelli)
+- `Role#members` had a typo preventing it from working ([#340](https://github.com/meew0/discordrb/pull/340))
+- `Message#my_reactions` now correctly returns `Array<Reaction>` ([#342](https://github.com/meew0/discordrb/pull/342))
+- Several internal checks have been added to make bots more resilient to zombie connections
+- Documentation for `TypingEvent` is now more accurate ([#367](https://github.com/meew0/discordrb/pull/367), thanks @Snazzah)
+- Corrected implementation of the `reason` parameter in various API ([#372](https://github.com/meew0/pull/372))
```
Comment (connorshea) [inline CHANGELOG.md]: This URL is missing the discordrb part, should be https://github.com/meew0/discordrb/pull/372

**Comment by z64** [comment]
cc @connorshea

File: CHANGELOG.md [hunk 10]
```diff
@@ -35,6 +35,34 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Support for category channels in `Server` and `Channel` ([#415](https://github.com/meew0/discordrb/pull/415), [#477](https://github.com/meew0/discordrb/pull/477), thanks @omnilord)
+- `CommandBot` and commands channel white-list can now be modified after creation ([#446](https://github.com/meew0/discordrb/pull/446), thanks @omnilord)
+- A `Role`'s `position` can now be sorted relative to other roles ([#445](https://github.com/meew0/discordrb/pull/445), thanks @mattantonelli)
+- The `return` keyword inside of commands can be used to return content to Discord ([#462](https://github.com/meew0/discordrb/pull/462), thanks @TrenchFroast)
+- `Emoji` now supplies `animated` ([#464](https://github.com/meew0/discordrb/pull/464), thanks @PixeLInc)
+- Additional instructions for installation of Ruby's devkit for Ruby 2.3+ ([#468](https://github.com/meew0/discordrb/pull/468), thanks @oct2pus)
+- `Server` API for retrieving a server's audit logs ([#353](https://github.com/meew0/discordrb/pull/353), thanks @Snazzah)
+- `EventContainer` methods for server role create, delete, and update events ([#494](https://github.com/meew0/discordrb/pull/494), thanks @Daniel-Worral, thanks @Daniel-Worrall)
```
Comment (connorshea) [inline CHANGELOG.md]: Double thanks

**Pull request closed without merging**